### PR TITLE
Evaluate lazily Android specific global values

### PR DESCRIPTION
**Pull request by jasssonpet**
Related to: https://github.com/NativeScript/NativeScript/issues/1563

ping @atanasovg, @KristinaKoeva

### CODE REVIEW

**Comment by enchev** [comment]
run ci

File: ui/transition/transition.android.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ export function _clearForwardTransitions(fragment: any): void {
         fragment[EXIT_POPENTER_TRANSITION] = undefined;
     }
 
-    if (_sdkVersion >= 21) {
+    if (_sdkVersion() >= 21) {
```
Comment (enchev) [inline ui/transition/transition.android.ts]: error TS2365: Operator '>=' cannot be applied to types '() => number' and 'number'.

Comment (jasssonpet) [inline ui/transition/transition.android.ts]: I'll take a look. It's compiling fine on my machine, though.

**Comment by atanasovg** [comment]
run ci

**Comment by hamorphis** [comment]
Sounds very neat.

**Comment by jasssonpet** [comment]
run ci

File: ui/transition/transition.android.ts (hunk 1 above)
Comment (jasssonpet) [inline ui/transition/transition.android.ts]: @enchev This is fixed now.

**Comment by vchimev** [comment]
run ci

**Comment by atanasovg** [comment]
:+1:

**Comment by PanayotCankov** [comment]
:+1:

**Comment by hshristov** [comment]
:+1:

File: utils/lazy.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function lazy<T>(action: () => T): () => T {
+    return (() => {
+        let _value: T;
+        return () => _value || (_value = action());
+    })();
+}
```
Comment (PanayotCankov) [inline utils/lazy.ts]: Can't we reuse the outer scope :D
``` TypeScript
export default function lazy<T>(action: () => T): () => T {
    let cache: T;
    return () => cache || (cache = action());
}
```

**Comment by hdeshev** [comment]
A premature optimization on my end - maybe we can move the `lazy` function to the "globals" module to avoid `require`-ing the module all over the place (the same thing we did for `moduleMerge`)

File: utils/lazy.ts (hunk 2 above)
Comment (jasssonpet) [inline utils/lazy.ts]: @PanayotCankov Yeah, the self-executing function is redundant.

**Comment by jasssonpet** [comment]
@hdeshev I like your suggestion, but let's push forward the webpack plugin. It will resolve such issues.

**Pull request merged**